bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/DiscoveryResultBuilder.java

```diff
@@ -36,7 +36,7 @@
     private final ThingUID thingUID;
 
     private @Nullable ThingUID bridgeUID;
-    private final Map<String, Object> properties = new HashMap<>();
+    private @Nullable Map<String, Object> properties;
```

Comment:
I would be careful here since it changes behavior of the builder. I know that 3.0.x addons must be recompiled to new packages, however such small changes remain undocumented and can blow up only when running a discovery with (most often) real things.

Reply:
You are right. I reverted it for the moment. Even if my change streamlined this builder with other OHC builders.